src/main/java/taxi/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,41 @@
+package taxi.web.filter;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class AuthenticationFilter implements Filter {
+    private Set<String> allowedUrls = new HashSet<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        allowedUrls.add("/drivers/login");
+        allowedUrls.add("/drivers/add");
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest request = (HttpServletRequest)servletRequest;
+        HttpServletResponse response = (HttpServletResponse)servletResponse;
+        HttpSession httpSession = request.getSession();
+        Long driverId = (Long) httpSession.getAttribute("driver_id");
+        if (driverId == null && allowedUrls.contains(request.getServletPath())) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+        if (driverId == null) {
+            response.sendRedirect("/drivers/login");
+            return;
```

Comment:
you can simplify too

Reply:
if (allowedUrl.contains(request.getServletPath()) || driverId != null) {
            filterChain.doFilter(request, response);
            return;
        }
        response.sendRedirect(request.getContextPath() + "/drivers/login");